pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -174,17 +175,20 @@ class ModelCheckpoint(Callback):
     """
 
     def __init__(self, filepath, monitor='val_loss', verbose=0,
-                 save_best_only=False, save_weights_only=False,
+                 save_top_k=0, save_weights_only=False,
```

Comment:
shouldn't this default to 1?

Reply:
it seems to me that `save_best_only=False` corresponds to `save_top_k=0`, when every model is saved. (since `period=1`)

Reply:
got it. so, save_top_k=0 saves all of them? I think we should use this notation:
0 = don't save anything
-1 = save all

Reply:
if that case, period option seems to be broken.

Reply:
added tests with desired behavioor